File: Practica 3/Pila.java
package PilasColas;
public class Pila {
    private int n;
    private int top;
    private long array[];;
    public Pila(int n){
        this.array = new long[n];
        this.top=-1;
        this.n=n;
    }
    public void push(long e){
        if(!isFull()){
            this.top++;
            this.array[top]=e;
        }
        else{
            System.out.println("Pila Llena!!!");
        }
    }
    public long pop(){
        if(!isEmpty()){
            long n = array[top];
            this.top--;
            return n;
        }
        else{
            System.out.println("Pila Vacia!!!");
        }
        return 0l;
    }
    public long peek(){
        if(!isEmpty()){
            long n = array[top];
            return n;
        }
        else{
            System.out.println("Pila Vacia!!!");
        }
        return 0l;
    }
    public boolean isEmpty(){
        if(this.top==-1){
            return true;
        }
        return false;
    }
    public boolean isFull(){
        if(this.top==n-1){
            return true;
        }
        return false;
    }
    public void mostrar(){
        for(int i=this.top;i>=0;i--){
            System.out.println(array[i]);
        }
    }
    public static void main(String[] args) {
        Pila p= new Pila(3);
        p.push(1l);
        p.push(2l);
        p.push(3l);
        p.mostrar();
        System.out.println("");
        System.out.println(p.pop());
        System.out.println("");
        p.mostrar();
        System.out.println("");
        System.out.println(p.peek());
        System.out.println("");
        p.mostrar();
    }
    
}

File: Practica 3/Pila.py
class Pila:
    def __init__(self, n):
        self.n = n
        self.top = -1
        self.array = [0] * n

    def push(self, e):
        if not self.isFull():
            self.top += 1
            self.array[self.top] = e
        else:
            print("Pila Llena!!!")

    def pop(self):
        if not self.isEmpty():
            n = self.array[self.top]
            self.top -= 1
            return n
        else:
            print("Pila Vacia!!!")
        return 0

    def peek(self):
        if not self.isEmpty():
            return self.array[self.top]
        else:
            print("Pila Vacia!!!")
        return 0

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1

    def mostrar(self):
        for i in range(self.top, -1, -1):
            print(self.array[i])

if __name__ == "__main__":
    p = Pila(3)
    p.push(1)
    p.push(2)
    p.push(3)
    p.mostrar()
    print("")
    print(p.pop())
    print("")
    p.mostrar()
    print("")
    print(p.peek())
    print("")
    p.mostrar()
